plugins/content/caption/base/lib/kCaptionsContentManager.php

```diff
@@ -307,14 +307,21 @@ public function createCaptionsFile($content, $clipStartTime, $clipEndTime, $time
 				{
 					$adjustedTimeLine = $this->createAdjustedTimeLine($matches, $clipStartTime, $clipEndTime, $globalOffset);
 					if($adjustedTimeLine)
-						$currentBlock .=$adjustedTimeLine;
+					{
+						$currentBlock .= $adjustedTimeLine;
+					}
 					else
+					{
 						$shouldAddBlockToNewFile = false;
+					}
 				}
 				else
+				{
 					$currentBlock .= $line . kCaptionsContentManager::UNIX_LINE_ENDING;
+				}
 				$line = kCaptionsContentManager::getNextValueFromArray($originalFileContentArray);
 			}
+			$shouldAddBlockToNewFile = $shouldAddBlockToNewFile && $currentBlock && trim($currentBlock) !== '';
```

Comment:
why needed?

Reply:
we divide the caption content to blocks, by new line, each block will have identifier (number), times and text
for example:
if we have a caption file with 2 new lines at the end then we will create an empty block (with identifier but no time and no text)